### PR TITLE
feat: remove RPC_PROVIDER_URL env

### PR DESCRIPTION
**Pull request by rndquu**
Right now we get RPC URL based on chain id from the bot's config https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L13 so we no longer need the `RPC_PROVIDER_URL` env variable

@0xcodercrane It seems that the `process.env.RPC_PROVIDER_URL` [env variable](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L27) exists in netlify so when chain id is set to gnosis the ethereum RPC URL is used (instead of gnosis RPC URL) so there is an error on permit generation. Could you remove the `RPC_PROVIDER_URL` env variable in netlify both for production and development deployments?

### CODE REVIEW

**Comment by 0xcodercrane** [comment]
> Right now we get RPC URL based on chain id from the bot's config https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L13 so we no longer need the `RPC_PROVIDER_URL` env variable
> 
> @0xcodercrane It seems that the `process.env.RPC_PROVIDER_URL` [env variable](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts#L27) exists in netlify so when chain id is set to gnosis the ethereum RPC URL is used (instead of gnosis RPC URL) so there is an error on permit generation. Could you remove the `RPC_PROVIDER_URL` env variable in netlify both for production and development deployments?

Yeah I can remove both of them. but I have to remove them after the PR merged no?

**Review by 0xcodercrane — COMMENTED** [review]
I think we might need to consider adding paid RPC endpoints as well

**Review by 0xcodercrane — APPROVED** [review]

File: .env.example
Comment (pavlovcik) [inline .env.example:18]:
```suggestion
EVM_PRIVATE_KEY=""
```

It probably makes sense to shorten because this is the only EVM private key of concern in this codebase, right?

**Comment by pavlovcik** [comment]
> I think we might need to consider adding paid RPC endpoints as well

What does "paid RPC endpoints" mean?

**Comment by 0xcodercrane** [comment]
> > I think we might need to consider adding paid RPC endpoints as well
> 
> What does "paid RPC endpoints" mean?

Like we purchased from alchemy/infura rpc providers.

**Comment by pavlovcik** [comment]
> > > I think we might need to consider adding paid RPC endpoints as well
> > 
> > 
> > What does "paid RPC endpoints" mean?
> 
> Like we purchased from alchemy/infura rpc providers.

I think we should eventually aim to set up our own endpoints if it's easy like we do on Cloudflare (its just a service they provide, for free.)

File: .env.example
Comment (rndquu) [reply .env.example:18]: The `UBIQUITY_BOT_EVM_PRIVATE_KEY` env variable is kind of deprecated. As soon as https://github.com/Alpha-Labs-Global team create an org wide config (with encrypted wallet private key) we are free to remove `UBIQUITY_BOT_EVM_PRIVATE_KEY` from our codebase and from netlify env variables.

Comment (pavlovcik) [reply .env.example]: Let's not wait on them.

Comment (rndquu) [reply .env.example]: It's been recently removed https://github.com/ubiquity/ubiquibot/pull/356

**Comment by rndquu** [comment]
@0xcodercrane could you:
1. merge this PR
2. remove the `RPC_PROVIDER_URL` env variable in netlify both for production and development deployments

**Comment by 0xcodercrane** [comment]
> @0xcodercrane could you:
> 
> 1. merge this PR
> 2. remove the `RPC_PROVIDER_URL` env variable in netlify both for production and development deployments

Sure can do

**Review by 0xcodercrane — APPROVED** [review]

**Comment by rndquu** [comment]
@0xcodercrane did you remove the `RPC_PROVIDER_URL` env variable from netlify (dev + prod)? waiting for it to check how payouts work in gnosis https://github.com/ubiquibot/production/issues/3

**Comment by 0xcodercrane** [comment]
> @0xcodercrane did you remove the `RPC_PROVIDER_URL` env variable from netlify (dev + prod)? waiting for it to check how payouts work in gnosis [ubiquibot/production#3](https://github.com/ubiquibot/production/issues/3)

You couldn't test it until we create a sync pr to main. so I am gonna remove `RPC_PROVIDER_URL` from dev. sounds ok?

**Comment by rndquu** [comment]
> > @0xcodercrane did you remove the `RPC_PROVIDER_URL` env variable from netlify (dev + prod)? waiting for it to check how payouts work in gnosis [ubiquibot/production#3](https://github.com/ubiquibot/production/issues/3)
> 
> You couldn't test it until we create a sync pr to main. so I am gonna remove `RPC_PROVIDER_URL` from dev. sounds ok?

Gnosis payouts are already deployed to production. So if you remove `RPC_PROVIDER_URL` both from prod and dev nothing breaks.

1. `RPC_PROVIDER_URL` is removed from prod and dev in netlify
2. [Here](https://github.com/ubiquity/ubiquibot/blob/main/src/bindings/config.ts#L27) the correct RPC is used (because we would have `process.env.RPC_PROVIDER_URL` undefined)

I'm trying to say that it is safe to remove `RPC_PROVIDER_URL` from prod and that we can omit waiting for a new bot's release. Gnosis payouts should start working as soon as  `RPC_PROVIDER_URL` is removed from netlify. But if you are concerned from removing `RPC_PROVIDER_URL` for prod then sure we can start from dev first.

**Comment by 0xcodercrane** [comment]
Removed it from both env

**Pull request merged**
